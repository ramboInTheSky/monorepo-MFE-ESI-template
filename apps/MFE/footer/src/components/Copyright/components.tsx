import styled from "styled-components"
import Typography from "@mui/material/Typography"
import Grid from "@mui/material/Grid"

export const CopyrightText = styled(Typography)<{component: "h3"}>`
    color: ${props => props.theme?.colours?.text?.muted || "#000"};
    display: block;
    text-align: center;
    font-size: 0.875rem;
    font-family: ${({theme}) => `${theme?.colours?.font?.regular?.family}, ${theme?.colours?.font?.default}`};
    letter-spacing: 0.12px;
    line-height: 1.5rem;
`

export const GridContainer = styled(Grid)`
    padding-top: 1.5rem;
    padding-bottom: 0.75rem;
`

export const ChildGrid = styled(Grid)`
    flex: 1 1 auto;
    padding: 0 0.75rem;
`
// more amperstands = high specificity
export const DeviceToggleLink = styled.a`
    &&& {
        font-size: 0.875rem;
        line-height: 1.5;
        margin-bottom: 0.75rem;
        display: inline-block;
        color: ${props => props.theme.colours.text.hyperlink};
        text-decoration: underline;
    }
`
