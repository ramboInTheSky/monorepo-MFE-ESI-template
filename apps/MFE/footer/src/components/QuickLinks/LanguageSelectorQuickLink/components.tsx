import styled from "styled-components"
import Grid from "@mui/material/Grid"
import Typography from "@mui/material/Typography"
import {breakpoints} from "@monorepo/themes"
import Link from "@mui/material/Link"

export const LanguageSelectorWrapper = styled("div")`
    display: flex;
    padding: 1rem;
    justify-content: space-between;
    align-items: center;
    height: 2.125rem;

    /* override existing css in mvc */
    box-sizing: content-box;
    @media (min-width: ${breakpoints.values.md}px) {
        padding-left: 2rem;
        padding-right: 2rem;
    }
    @media (min-width: ${breakpoints.values.lg}px) {
        padding: 0.75rem 1.125rem;
        height: 42px;
    }
`

export const LanguageSelectorImg = styled("img")`
    padding: 0.25rem 0;
    vertical-align: middle;
    display: inline;
`

export const LanguageSelectorElement = styled(Grid)`
    span {
        display: inline-block;
    }
    a,
    span {
        font-size: 0.875rem;
        text-transform: capitalize;
    }
`

export const LanguageSelectorLink = styled(Link)`
    &&&.altLanguageLink {
        display: inline-block;
        padding: 0;
        color: ${props => props.theme.colours.text.hyperlink};

        &:active,
        &:hover,
        &:focus,
        &:visited {
            color: ${props => props.theme.colours.text.hyperlink};
        }
    }
`

export const LanguageSelectorTitle = styled(Typography)`
    &&& {
        display: inline-block;
        margin: 0 0.25rem 0 1.125rem;
    }
`

export const LanguageSelectorDescription = styled(Typography)`
    display: inline-block;
`
