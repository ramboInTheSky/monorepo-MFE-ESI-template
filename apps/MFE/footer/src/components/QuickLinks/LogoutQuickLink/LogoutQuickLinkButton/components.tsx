import styled from "styled-components"

export const LogoutLinkComponent = styled("a")`
    width: 88px;
    height: 24px;
    background: white;
    font-family: "AzoSansMedium", -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial,
        Noto Sans, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
    font-size: 12px;
    letter-spacing: 1px;
    padding: 4px;
    border: 1px solid #000;
    border-radius: 4px;
    text-transform: uppercase;
    font-weight: 500;
    display: inline-block;
    text-align: center;
    box-sizing: border-box;
    text-decoration: none;
    color: #000;
`
