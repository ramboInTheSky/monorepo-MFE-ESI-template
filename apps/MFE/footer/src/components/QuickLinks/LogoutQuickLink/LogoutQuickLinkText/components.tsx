import Typography from "@mui/material/Typography"

import styled from "styled-components"

export const LogoutText = styled(Typography)`
    /* Override MVC  */
    text-transform: capitalize;
    padding-bottom: 0;
`
