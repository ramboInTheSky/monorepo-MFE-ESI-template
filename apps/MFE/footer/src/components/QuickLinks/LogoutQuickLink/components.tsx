import styled from "styled-components"

import Grid from "@mui/material/Grid"
import Hidden from "@mui/material/Hidden"

export const LogoutHidden = styled(Hidden)`
    width: 100%;
`

export const LogoutWrapper = styled.div`
    display: inline-block;
    margin-right: 0.75rem;
    padding: 1rem 0;
`

export const LogoutContainer = styled(Grid)`
    border-bottom: ${props => props.theme.colours.utilities.divider};
`
