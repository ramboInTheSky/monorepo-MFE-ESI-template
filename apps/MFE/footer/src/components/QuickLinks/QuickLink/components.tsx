import styled from "styled-components"
import Typography from "@mui/material/Typography"
import Link from "@mui/material/Link"

export const StyledLink = styled(Link)`
    /* Override MVC  */
    && {
        &:visited {
            color: ${props => props.theme.colours.text.default};
        }
    }
`

export const QuickLinkContent = styled("span")`
    display: inline-block;
    margin-left: 1.125rem;
    margin-right: 0;
`

export const QuickLinkTitle = styled(Typography)`
    /* Override MVC  */

    && {
        text-transform: capitalize;
        padding-bottom: 0;
    }
`
export const QuickLinkDescription = styled(Typography)<{component: "h4"}>`
    /* Priority over parent element (Link element) */
    && {
        color: ${props => props.theme.colours.text.muted};
    }
`

export const QuickLinkImg = styled.img`
    width: 1.5rem;
    height: 1.5rem;
`
