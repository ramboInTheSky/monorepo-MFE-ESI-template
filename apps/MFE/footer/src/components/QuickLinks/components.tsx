import styled from "styled-components"
import Grid from "@mui/material/Grid"
import {breakpoints} from "@monorepo/themes"

export const GridContainer = styled(Grid)`
    @media (min-width: ${breakpoints.values.md}px) {
        div:last-child {
            border-left: 0;
            border-right: inherit;
        }
        div.quicklink__wrapper:first-child > a {
            padding-left: 2rem;
            padding-right: 1.125rem;
        }
    }
    @media (min-width: ${breakpoints.values.md}px) and (max-width: ${breakpoints.values.lg - 1}px) {
        div.quicklink__wrapper:nth-child(2n + 2) {
            border-right: 0;
        }
        div.quicklink__wrapper:nth-child(2n + 1) > a {
            padding-left: 2rem;
            padding-right: 1.125rem;
        }
    }
    @media (min-width: ${breakpoints.values.lg}px) {
        div.quicklink__wrapper:last-child {
            border-bottom: 0;
        }
    }
    @media (min-width: ${breakpoints.values.xl}px) {
        div.quicklink__wrapper:first-child > a {
            padding-left: 2.5rem;
            padding-right: 1.125rem;
        }
    }
`

export const QuickLinksWrapper = styled(Grid)`
    border-bottom: ${props => props.theme.colours.utilities.divider};

    @media (min-width: ${breakpoints.values.md}px) {
        border-right: ${props => props.theme.colours.utilities.divider};
    }
    @media (min-width: ${breakpoints.values.lg}px) {
        border-bottom: 0;
    }

    a {
        padding: 0.75rem 1rem;
        display: flex;
        align-items: center;
    }
`
