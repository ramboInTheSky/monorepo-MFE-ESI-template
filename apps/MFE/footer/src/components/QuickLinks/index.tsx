import React, {Component} from "react"
import {GridSize} from "@mui/material"
import {formatTextTestIds} from "@monorepo/utils"
import {RegionModel, SubRegionElementModel} from "../../models/footerModel"
import {DefaultColumns} from "../../models/regions"
import getwindow from "../../utils/window"
import {UserDuckState, AccountStatusChanged} from "../../ducks/user"
import QuickLink from "./QuickLink"
import MyAccountQuickLink from "./MyAccountQuickLink"
import LogoutLinkComponent from "./LogoutQuickLink"
import LanguageSelectorQuickLink from "./LanguageSelectorQuickLink"
import {QuickLinksElementsTransformer, QuickLinkTypes} from "./utils"
import connect from "./connect"
import {QuickLinksWrapper, GridContainer} from "./components"
import executeOnLoad from "../../utils/window/executeOnLoad"

export interface QuickLinksProps {
    region: RegionModel
    accountStatusChanged: AccountStatusChanged
    user: UserDuckState
    showLangSelector: boolean
}

enum SupportedQuickLinksTypes {
    Logout = "Logout",
}

export class QuickLinks extends Component<QuickLinksProps> {
    window

    componentDidMount = () => {
        this.window = getwindow()
        this.userDataChanged()

        executeOnLoad(this.subscribeToAmidoBasket)
    }

    componentWillUnmount = () => {
        this.window.removeEventListener("load", this.subscribeToAmidoBasket)
    }

    subscribeToAmidoBasket = () => {
        this.userDataChanged()

        if (this.window.AmidoBasket) {
            this.window.AmidoBasket.RegisterChange(() => {
                this.userDataChanged()
            })
        }
    }

    userDataChanged = () => {
        const {AmidoBasket} = this.window
        let firstName
        if (AmidoBasket && AmidoBasket.Data && AmidoBasket.Data.FirstName) firstName = AmidoBasket.Data.FirstName
        if (firstName !== this.props.user.accountFirstName) this.props.accountStatusChanged(firstName)
    }

    renderDefaultQuickLinks = (elements: SubRegionElementModel[]) => {
        const condensedElements = QuickLinksElementsTransformer(elements)
        let mdcolumns = 6
        const languageSelectorElement = condensedElements?.find(
            element => element.type === QuickLinkTypes.LanguageSelector,
        )
        const islanguageselectorpresent = !!languageSelectorElement
        if (islanguageselectorpresent) {
            mdcolumns = 6
        }
        const numberOfStandardLinks = condensedElements?.filter(
            element => element.type === QuickLinkTypes.Standard || element.type === QuickLinkTypes.MyAccount,
        )

        return condensedElements?.map((quickLink: any) => {
            const {url, text, accessibilityText, type} = quickLink

            if (
                (type !== QuickLinkTypes.MyAccount &&
                    type !== QuickLinkTypes.LanguageSelector &&
                    (!text || !url || !accessibilityText)) ||
                (quickLink.type === QuickLinkTypes.LanguageSelector && !this.props.showLangSelector)
            ) {
                return null
            }

            const columnStandardLength: GridSize =
                numberOfStandardLinks && Object.keys(numberOfStandardLinks).length === 1 ? 6 : 3

            const columns = {
                sm: 12,
                md: mdcolumns,
                lg: type === QuickLinkTypes.LanguageSelector ? 5 : columnStandardLength,
            }

            return (
                <QuickLinksWrapper
                    data-testid={formatTextTestIds(`footer-quicklinks-${type}-${text}`)}
                    key={`${type}-${text}`}
                    item
                    xs={DefaultColumns.xs}
                    sm={columns.sm as GridSize}
                    md={columns.md as GridSize}
                    lg={columns.lg as GridSize}
                    className="quicklink__wrapper"
                >
                    {quickLink.type === QuickLinkTypes.MyAccount && <MyAccountQuickLink {...quickLink} />}
                    {quickLink.type === QuickLinkTypes.LanguageSelector && (
                        <LanguageSelectorQuickLink data={quickLink} />
                    )}
                    {quickLink.type === QuickLinkTypes.Standard && <QuickLink data={quickLink} />}
                </QuickLinksWrapper>
            )
        })
    }

    render() {
        const {subRegions} = this.props.region
        let key = 0
        return (
            <>
                {subRegions.map(subregion => {
                    const currentKey = key
                    key += subregion.elements.length
                    switch (subregion.title) {
                        case SupportedQuickLinksTypes.Logout:
                            return (
                                <GridContainer key={`${subregion.title}-${key}`} item container xs={DefaultColumns.xs}>
                                    <LogoutLinkComponent elements={subregion.elements} key={currentKey} />
                                </GridContainer>
                            )
                        case "":
                            return (
                                <GridContainer key={`${subregion.title}-${key}`} item container xs={DefaultColumns.xs}>
                                    {this.renderDefaultQuickLinks(subregion.elements)}
                                </GridContainer>
                            )
                        default:
                            return null
                    }
                })}
            </>
        )
    }
}

export default connect(QuickLinks)
