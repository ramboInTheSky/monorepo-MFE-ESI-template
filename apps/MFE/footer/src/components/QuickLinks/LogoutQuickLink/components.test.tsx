import React from "react"
import {render} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"
import {mockTheme} from "@monorepo/themes"
import {LogoutWrapper, LogoutHidden, LogoutContainer} from "./components"

describe("LogoutQuickLink: ", () => {
    describe("LogoutWrapper: ", () => {
        it("should match the snapshot ", () => {
            const {asFragment} = render(<LogoutWrapper />)
            expect(asFragment()).toMatchSnapshot()
        })
    })
    describe("LogoutHidden: ", () => {
        it("should match the snapshot ", () => {
            const {asFragment} = render(<LogoutHidden />)
            expect(asFragment()).toMatchSnapshot()
        })
    })
    describe("LogoutContainer: ", () => {
        it("should match the snapshot ", () => {
            const {asFragment} = render(
                <SCThemeProvider theme={mockTheme}>
                    <LogoutContainer />
                </SCThemeProvider>,
            )
            expect(asFragment()).toMatchSnapshot()
        })
    })
})
