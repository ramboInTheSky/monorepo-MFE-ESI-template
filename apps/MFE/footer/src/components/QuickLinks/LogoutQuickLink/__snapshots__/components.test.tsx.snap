// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutQuickLink:  LogoutContainer:  should match the snapshot  1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root components__LogoutContainer-sc-19qf9d4-2 iGbDKo"
  />
</DocumentFragment>
`;

exports[`LogoutQuickLink:  LogoutHidden:  should match the snapshot  1`] = `<DocumentFragment />`;

exports[`LogoutQuickLink:  LogoutWrapper:  should match the snapshot  1`] = `
<DocumentFragment>
  <div
    class="components__LogoutWrapper-sc-19qf9d4-1 kehuth"
  />
</DocumentFragment>
`;
